hs_core/views/discovery_json_view.py

```diff
@@ -0,0 +1,41 @@
+import simplejson as json
+from django.http import HttpResponse
+from haystack.query import SearchQuerySet
+from django import forms
+from haystack.forms import FacetedSearchForm
+from haystack.generic_views import FacetedSearchView
+from django.core import serializers
+from hs_core.discovery_form import DiscoveryForm
+
+class DiscoveryJsonView(FacetedSearchView):
+    facet_fields = ['author', 'subjects', 'resource_type', 'public', 'owners_names', 'discoverable']
+    form_class = DiscoveryForm
+
+    def form_valid(self, form):
+        coor_values = []
+        coordinate_dictionary = []
+        self.queryset = form.search()
+        if len(self.request.GET):
+            for result in self.get_queryset():
+                json_obj = {}
+                json_obj['title'] = result.object.title
+                json_obj['get_absolute_url'] = result.object.get_absolute_url()
+                for coverage in result.object.metadata.coverages.all():
```

Comment:
Haystack is capable of storing fields, both those which are indexed and additional fields, which you can then access from the result objects that come back from your call to `get_queryset()`. The data needed for this coverage can be added to the resource index, and then read from the results. This will save the need to do multiple database queries to Postgres in addition to the Solr query you've already done for the results.